game/scripts/vscripts/abilities/siltbreaker/siltbreaker_protection.lua

```diff
@@ -0,0 +1,19 @@
+
+function siltbreaker_protection_purge(kv)
+  local unit = kv.unit
+  if unit:IsStunned() or unit:IsSilenced() then
+    unit:Purge( false, true, true, true, true )
+  end
+end
+
+function siltbreaker_protection_trigger(kv)
+  local unit = kv.unit
+  local ability = kv.ability
+  local protection_duration = ability:GetSpecialValueFor("duration")
+
+  if ability:IsCooldownReady() and (unit:IsStunned() or unit:IsSilenced()) then
+    unit:Purge( false, true, true, true, true )
```

Comment:
bFrameOnly true? I'm sure it works, but why restrict it?

Reply:
I don't event know what it does. Should I remove it?

Reply:
Restricts the purge to only remove modifiers created on that frame. So like only purges stuff that was applied immediately before the purge. Though I'm not sure it actually works like it's supposed to. I don't think anything in vanilla Dota works that way.